### ISSUE TITLE
Azure Batch instance types automatic update

### ISSUE THREAD

**Issue by pditommaso** [nextflow-io/nextflow#2994]
The Azure Batch support implemented by nextflow is taking the machine-type and location information from set of files embedded in the application itself. 

https://github.com/nextflow-io/nextflow/blob/master/plugins/nf-azure/src/resources/nextflow/cloud/azure#L63-L63

However this is suboptimala because this information changes overtime and it's quite a hassle to be maintained manually 

The goal of this issue is to implement an automatic update system that fetches this information from an Azure endpoint, and cache in the launching environment local disk. 

The flow should be the following: 
1. check if the `$NXF_HOME/azure/cache/instance-info.json` exists 
2. if the file exists and it's not expired (older than 7 days) use it as a replacement of the current instance type and locations DB 
3. if the file does not exist or it's expired download the instance types endpoint, the continue at point 2
4. if for any reason it cannot be downloaded fallback on the current database

#### Caveat 

* this feature should work with the existing Batch credentials i.e. the user should not be required for new keys to access those info

**Comment by abhi18av** [comment]
@pditommaso I have found a solution for https://github.com/nextflow-io/nextflow/issues/2994 , however it relies on the use of subscription ID and an authenticated client, which can been added via https://github.com/nextflow-io/nextflow/pull/3132 


TLDR;
- The instance types can differ for an account/region/tier (for which we maintain multiple JSON files)
- We are using our own subscription ID https://github.com/nextflow-io/nextflow/blob/master/plugins/nf-azure/src/resources/nextflow/cloud/azure/az-locations.json#L4 for generating this list
- Using service-principal based setup, users can specify their own subscription ID to fetch the list
